docs/dapp/opl/build.md

```diff
@@ -0,0 +1,159 @@
++    if (!('url' in hostConfig)) throw new Error(`${args.hostNetwork} not configured`);
++    const provider = new ethers.providers.JsonRpcProvider(hostConfig.url);
++    let nonce = await provider.getTransactionCount(signerAddr);
++    if (args.hostNetwork === 'local') nonce++;
++    const daoAddr = ethers.utils.getContractAddress({ from: signerAddr, nonce });
++
++    const ballotBox = await BallotBoxV1.deploy(daoAddr);
++    await ballotBox.deployed();
++    console.log('expected DAO', daoAddr);
++    console.log('BallotBox', ballotBox.address);
++    return ballotBox.address;
++  });
++
++task('deploy-dao')
++  .addParam('ballotBoxAddr')
++  .setAction(async (args, hre) => {
++    await hre.run('compile');
++    const DAOV1 = await hre.ethers.getContractFactory('DAOV1');
++    const dao = await DAOV1.deploy(args.ballotBoxAddr);
++    await dao.deployed();
++    console.log('DAO', dao.address);
++    return dao;
++  });
++
++task('deploy-local').setAction(async (_args, hre) => {
++    await hre.run('compile');
++    const ballotBox = await hre.run('deploy-ballot-box', { hostNetwork: 'local' });
++    await hre.run('deploy-dao', { ballotBoxAddr: ballotBox });
++  });
+ 
+ const config: HardhatUserConfig = {
+   solidity: "0.8.18",
++  networks: {
++    hardhat: {
++      chainId: 1337, // @see https://hardhat.org/metamask-issue.html
++    },
++    local: {
++      url: 'http://127.0.0.1:8545',
++    },
++    'bsc-testnet': {
++      url: 'https://data-seed-prebsc-1-s1.binance.org:8545',
++      chainId: 97,
++      accounts,
++    },
++    'sapphire-testnet': {
++      url: 'https://testnet.sapphire.oasis.dev',
++      chainId: 0x5aff,
++      accounts,
++    },
++  }
+ };
+ 
+ export default config;
+```
+
+#### Localhost
+
+We can start local Hardhat node again:
+
+```sh
+npx hardhat node
+```
+
+Our deploy should succeed locally.
+
+```sh
+npx hardhat deploy-local --network localhost
+Nothing to compile
+No need to generate any newer typings.
+Nothing to compile
+No need to generate any newer typings.
+expected DAO 0xa513E6E4b8f2a923D98304ec87F64353C4D5C853
+BallotBox 0x0165878A594ca255338adfa4d48449f69242Eb8F
+Nothing to compile
+No need to generate any newer typings.
+DAO 0xa513E6E4b8f2a923D98304ec87F64353C4D5C853
+```
+
+We will use these addresses in our frontend application.
+
+#### Testnet
+
+We can likewise deploy to OPL [testnet](../../dapp/sapphire/guide#testnet-and-mainnet) with Sapphire.
+
+In this case, we should prepare a wallet with testnet tokens on both BNB Smart
+Chain [Faucet](https://testnet.bnbchain.org/faucet-smart) and Sapphire [faucet](https://faucet.testnet.oasis.dev).
+
+Set the private key.
+
+```sh
+export PRIVATE_KEY=
+```
+
+And deploy with the testnet parameters.
+
+```sh
+npx hardhat deploy-ballot-box --network sapphire-testnet --host-network bsc-testnet
+```
+
+Then, use the ballox box address below
+
+```sh
+npx hardhat deploy-dao --network bsc-testnet --ballot-box-addr {BALLOT_BOX_ADDR}
+```
```

Comment:
```suggestion
```

:::info Example

You can explore and download a [complete example] from the Oasis Playground repository.

:::

[complete example]: https://github.com/oasisprotocol/playground/tree/main/opl-secret-ballot
```